mapstr.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+)
+
+// stringifyKeysMapValue recurses into v and changes all instances of
+// map[interface{}]interface{} to map[string]interface{}. This is useful to
+// work around the impedence mismatch between JSON and YAML unmarshaling that's
+// described here:
+//
+// https://github.com/go-yaml/yaml/issues/139
+func stringifyKeysMapValue(v interface{}) interface{} {
+	switch v := v.(type) {
+	case []interface{}:
+		return stringifyKeysInterfaceArray(v)
+	case map[interface{}]interface{}:
+		return stringifyKeysInterfaceMap(v)
+	default:
+		return v
+	}
+}
+
+//
+// helpers
+//
+
+func stringifyKeysInterfaceArray(in []interface{}) []interface{} {
+	res := make([]interface{}, len(in))
+	for i, v := range in {
+		res[i] = stringifyKeysMapValue(v)
+	}
+	return res
+}
+
+func stringifyKeysInterfaceMap(in map[interface{}]interface{}) map[string]interface{} {
+	res := make(map[string]interface{})
+	for k, v := range in {
+		res[fmt.Sprintf("%v", k)] = stringifyKeysMapValue(v)
```

Comment:
I think we should check that the key is already a string and raise an error if it isn't. Otherwise, this can lead to some really strange behavior. Suppose the original YAML file looks like:
```
in: a
on: b
under: c
```
The YAML parser will interpret `in` and `under` as strings but it will interpret `on` as the boolean `true`. So the output of `stringifyKeysMapValue` will be:
```
map[string]interface{}{
  "in": "a",
  "true": "b",
  "under": "c",
}
```
which is super confusing and almost certainly not what was intended.

Reply:
Great point. I adapted this code from elsewhere, but yep, in our case I think it makes perfect sense to be as loud as possible about this problem because we expect it to _never_ happen.

Patched in 24a2bb4. Thanks for taking a look.